Payroll account configuration

lomamech opened the pull request:
- Implement client side for Payroll Accounting Configuration
- Implement API for Payroll Accounting Configuration
- Implement integration test for Payroll Account Configuration
- Implement E2E test for Payroll Account Configuration

reference #2292

lomamech committed:
feature(Payroll Account Configuration)
- Implement client side for Payroll Accounting Configuration
- Implement API for Payroll Accounting Configuration

lomamech committed:
test(Payroll Account Configuration)
- Implement integration test for Payroll Account Configuration

lomamech committed:
test(Payroll Account Configuration)
-Implement E2E test for Payroll Account Configuration

jniles commented on client/src/modules/payroll/account_configuration/configuration.service.js:
```diff
@@ -0,0 +1,17 @@
+ConfigurationAccountService.$inject = ['PrototypeApiService', '$uibModal'];
+
+/**
+ * @class RubricService
```
These comments need to be updated.

lomamech committed:
Fix test integration remove status only in account configuration

jniles commented on client/src/modules/payroll/account_configuration/modals/account.modal.js:
```diff
@@ -0,0 +1,59 @@
+  '$state', 'ConfigurationAccountService', 'ModalService', 'NotifyService', 'appcache',
+];
+
+function AccountConfigModalController($state, Config, ModalService, Notify, AppCache) {
```
It doesn't look like you use `ModalService` here, could you remove it?

jniles commented on client/src/modules/payroll/account_configuration/templates/account.tmpl.html:
```diff
@@ -0,0 +1,3 @@
+<div class="ui-grid-cell-contents text-left">
```
The ui-grid templates are usually pretty heavy to load.  Could you refactor this to simply render a value?  It would be much more performant.   Like this:
```js
.then(rows => {
  rows.forEach(row => row.hrAccountText = `[${row.account_number}] ${row.account_label}`);
  // etc...
})


// In your column configuration, you can now just use the `hrAccountText` column!
```

jniles commented on client/src/modules/payroll/account_configuration/templates/action.tmpl.html:
```diff
@@ -0,0 +1,22 @@
+    <span class="caret"></span>
+  </a>
+
+  <ul data-action="{{rowRenderIndex}}"  class="dropdown-menu-right" bh-dropdown-menu-auto-dropup uib-dropdown-menu>
```
Nice autodropup 👍

jniles commented on server/controllers/payroll/accounts/index.js:
```diff
@@ -0,0 +1,112 @@
+* This controller exposes an API to the client for reading and writing Account
+*/
+
+var db = require('../../../lib/db');
```
For these variables, it is much better to use `const`.

jniles commented on client/src/i18n/en/tree.json:
```diff
@@ -2,6 +2,7 @@
   "TREE": {
     "ACCOUNT": "Account Management",
     "ACCOUNT_STATEMENT": "Account Statement",
+    "ACCOUNT_CONFIGURATION":"Account Configuration",
```
Could you call the key `PAYROLL_ACCOUNT_CONFIGURATION`?  We might get confused about what it is for otherwise...

jniles commented on client/src/modules/payroll/account_configuration/configuration.js:
```diff
@@ -0,0 +1,97 @@
+    fastWatch         : true,
+    flatEntityAccess  : true,
+    enableSorting     : true,
+    onRegisterApi     : onRegisterApiFn,
```
👍

jniles reviewed (commented):
Looks pretty good to me!  A few suggestions to increase clarity and performance.

Since we now compile a lot of our code to ES2015+, please use the new JS methods as much as possible.  Things like arrow functions and `let`/`const` should be used wherever you have an anonymous function or use a `var`.

lomamech committed:
refactor(Payroll Account Configuration)
- Use Payroll_account_configuration instead Account_configuration
- Sanitaze Code
- Remove not necessary account.tmpl
- Use const instead var and remplace some var with let

jniles commented on client/src/modules/payroll/account_configuration/configuration.js:
```diff
@@ -0,0 +1,96 @@
+  var gridColumn =
+    [
+      { field : 'label', displayName : 'FORM.LABELS.DESIGNATION', headerCellFilter : 'translate' },
+      { field : 'hrAccountText', displayName : 'FORM.LABELS.ACCOUNT', headerCellFilter : 'translate' },
```
Nice!

jniles reviewed (approved):
Looks good to me.

bors r+
